```
builder for '/nix/store/kp5qmxrq5rvpiqy42830lbmzbshfgz5q-python3.8-isort-4.3.21.drv' failed with exit code 1; last 10 log lines:
  >           assert str(tmpdir.join("file1.py")) in out
  E           AssertionError: assert '/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0/file1.py' in 'Fixing /private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0/file2.py\n'
  E            +  where '/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0/file1.py' = str(local('/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0/file1.py'))
  E            +    where local('/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0/file1.py') = <bound method LocalPath.join of local('/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0')>('file1.py')
  E            +      where <bound method LocalPath.join of local('/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0')> = local('/private/tmp/pytest-of-siraben/pytest-2/test_command_line_True_0').join
  
  /private/tmp/nix-build-python3.8-isort-4.3.21.drv-0/isort-4.3.21/test_isort.py:2759: AssertionError
  =========================== short test summary info ============================
  FAILED test_isort.py::test_command_line[True] - AssertionError: assert '/priv...
  ================= 1 failed, 153 passed, 2 deselected in 5.57s ==================
cannot build derivation '/nix/store/0nsqjf5b1hk26syc33k0wr9zm0ygabqq-python3.8-poetry-core-1.0.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/1jahrni7z0x2zr4pii0ajyjkvhl637hw-python3.8-validators-0.17.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/2acjmf3z78wxw6yw71g5wp8jgsdf915m-python3.8-poetry-1.1.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/kjf1vfwmymg79q6mnxg0c50pn054b62w-python3.8-tinydb-4.1.1.drv': 1 dependencies couldn't be built
builder for '/nix/store/bx8w80rv5gma0pkkvjr3afic7zjxkaaq-python3.8-pypandoc-1.5.drv' failed with exit code 1; last 10 log lines:
  ?       -
  + # üäöîôû? ++
  - ======
  
  ----------------------------------------------------------------------
  Ran 24 tests in 8.572s
  
  FAILED (failures=1, skipped=1)
  Test failed: <unittest.runner.TextTestResult run=24 errors=0 failures=1>
  error: Test failed: <unittest.runner.TextTestResult run=24 errors=0 failures=1>
cannot build derivation '/nix/store/fw65fhfd2kj608zswfhyw1nk8ai1jb5x-archivy-0.8.4.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/5kf5i03yg4bfqf5ak3diprkbrcq6w6c9-env.drv': 1 dependencies couldn't be built
```

--- a/pkgs/applications/misc/archivy/default.nix
+++ b/pkgs/applications/misc/archivy/default.nix
@@ -1,0 +1,57 @@
+{ stdenv, lib, python3, fetchPypi, appdirs, attrs, requests,
+beautifulsoup4, click-plugins, elasticsearch, flask_login, flask_wtf,
+pypandoc, python-dotenv, python-frontmatter, tinydb, validators,
+watchdog, wtforms }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "archivy";
+  version = "0.8.5";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "144ckgxjaw29yp5flyxd1rnkm7hlim4zgy6xng7x0a9j54h527iq";
+  };
+
+  # Relax some dependencies
+  postPatch = ''
+    substituteInPlace requirements.txt \
+      --replace 'validators ==' 'validators >=' \
+      --replace 'elasticsearch ==' 'elasticsearch >=' \
+      --replace 'python-dotenv ==' 'python-dotenv >=' \
+      --replace 'beautifulsoup4 ==' 'beautifulsoup4 >=' \
+      --replace 'WTForms ==' 'WTForms >=' \
+      --replace 'python_dotenv ==' 'python_dotenv >=' \
+      --replace 'attrs == 20.2.0' 'attrs' \
+      --replace 'python_frontmatter == 0.5.0' 'python_frontmatter' \
+      --replace 'requests ==' 'requests >='
+  '';
+
+  propagatedBuildInputs = [
+    appdirs
+    attrs
+    beautifulsoup4
+    click-plugins
+    elasticsearch
+    flask_login
+    flask_wtf
+    pypandoc
+    python-dotenv
+    python-frontmatter
+    tinydb
+    requests
+    validators
+    watchdog
+    wtforms
+  ];
+
+  # __init__.py attempts to mkdir in read-only file system
+  doCheck = false;
+
+  meta = with stdenv.lib; {
+    description = "Self-hosted knowledge repository";
+    homepage = "https://archivy.github.io";
+    license = licenses.mit;
+    maintainers = with maintainers; [ siraben ];
+    platforms = platforms.unix;
+  };
+}

--- a/pkgs/development/python-modules/python-frontmatter/default.nix
+++ b/pkgs/development/python-modules/python-frontmatter/default.nix
@@ -1,0 +1,30 @@
+{ stdenv, lib, fetchFromGitHub, python3Packages }:
+
+python3Packages.buildPythonPackage rec {
+  pname = "python-frontmatter";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner = "eyeseast";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1iki3rcbg7zs93m3mgqzncybqgdcch25qpwy84iz96qq8pipfs6g";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    pyyaml
+    six
+  ];
+
+  checkInputs = with python3Packages; [
+    pytest
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/eyeseast/python-frontmatter";
+    description = "Parse and manage posts with YAML (or other) frontmatter";
+    license = licenses.mit;
+    maintainers = with maintainers; [ siraben ];
+    platforms = lib.platforms.unix;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -849,6 +849,8 @@ in
 
   archivemount = callPackage ../tools/filesystems/archivemount { };
 
+  archivy = python3Packages.callPackage ../applications/misc/archivy { };
+
   arandr = callPackage ../tools/X11/arandr { };
 
   inherit (callPackages ../servers/nosql/arangodb {

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5848,6 +5848,8 @@ in {
 
   python-forecastio = callPackage ../development/python-modules/python-forecastio { };
 
+  python-frontmatter = callPackage ../development/python-modules/python-frontmatter { };
+
   python-gitlab = callPackage ../development/python-modules/python-gitlab { };
 
   python-gnupg = callPackage ../development/python-modules/python-gnupg { };

